src/photon/photon_scheduler.c

```diff
@@ -133,7 +133,14 @@ void process_plasma_notification(event_loop *loop,
   local_scheduler_state *s = context;
   /* Read the notification from Plasma. */
   object_id obj_id;
-  recv(client_sock, (char *) &obj_id, sizeof(obj_id), 0);
+  int error = read_bytes(client_sock, (uint8_t *) &obj_id, sizeof(obj_id));
+  if (error == -1) {
```

Comment:
I would make this < 0 . The convention is to either return -1 (and set errno) or return -errno. The < 0 condition would be general enough for both (should we ever change the read_bytes in the future.